development/tools/pkg/file/file.go

```diff
@@ -38,3 +38,21 @@ func ReadFile(filePath string) (string, error) {
 
 	return content, nil
 }
+
+// SaveDataToTmpFile writes a slice of bytes to a tmp file
+func SaveDataToTmpFile(artifactData []byte, tmpFilePattern string) (*os.File, error) {
```

Comment:
The function would work as well if you provide it random bytes, not artifacts. I would change the signature to 
````SaveToTmpFile(data []byte, tmpFilePattern string) (*os.File, error)````
 Little bit shorter and more generic.